packages/@ember/application/index.ts

```diff
@@ -30,6 +30,16 @@ import RouterService from '@ember/routing/router-service';
 import type { EngineInstanceOptions } from '@ember/engine/instance';
 import type { SimpleDocument, SimpleElement } from '@simple-dom/interface';
 
+/**
+ * @deprecated Use `import { getOwner } from '@ember/owner';` instead.
```

Comment:
Should we runtime deprecate this function as well?

Reply:
Yes, but I wanted to land that separately from this—both because it's safer and keeps the types/behavior separately, and also (perhaps more importantly) because we usually try not to turn on the deprecation in the same release as we put out the replacement for it, because otherwise people get hammered with deprecations they haven't had a chance to do fix, including from addons.